com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -810,60 +810,60 @@ define([
-                                job_contract_ids.push(data.values[i]['id']);
-
-                                var contract = {
-                                    id: data.values[i]['id'],
-                                    title: data.values[i]['title'],
-                                    start_date: data.values[i]['period_start_date'],
-                                    end_date: data.values[i]['period_end_date'],
-                                    status: status,
-                                    revisions: data.values[i]['revisions']
-                                };
-
-                                var optionalEndDate = $filter('formatDate')(contract.end_date) || 'Unspecified';
-                                contract.label = contract.title + ' (' + $filter('formatDate')(contract.start_date) + ' - ' + optionalEndDate + ')';
-
-                                contractsData[data.values[i]['id']] = contract;
-                            }
-
-                            // Store the ContractsData what we can reuse later
-                            job_roles.contractsData = contractsData;
+            function contractIdsFromContact(contactId) {
+                return HRJobRolesService.getContracts(contactId).then(function (data) {
+                    var jobContractIds = [];
+                    var contractsData = {};
+
+                    // If we have job contracts, try to get the job roles for the contract
+                    if (data.count !== 0) {
+                        for (var i = 0; i < data.count; i++) {
+                            // Job contract IDs which will be passed to the "getAllJobRoles" service
+                            jobContractIds.push(data.values[i]['id']);
+
+                            var contract = {
+                                id: data.values[i]['id'],
+                                title: data.values[i]['title'],
+                                start_date: data.values[i]['period_start_date'],
+                                end_date: data.values[i]['period_end_date'],
+                                status: status,
+                                revisions: data.values[i]['revisions']
+                            };
+
+                            var optionalEndDate = $filter('formatDate')(contract.end_date) || 'Unspecified';
+                            contract.label = contract.title + ' (' + $filter('formatDate')(contract.start_date) + ' - ' + optionalEndDate + ')';
+
+                            contractsData[data.values[i]['id']] = contract;
+                        }
 
-                            job_roles.job_contract_ids = job_contract_ids;
+                        // Store the ContractsData what we can reuse later
+                        job_roles.contractsData = contractsData;
+                        job_roles.job_contract_ids = jobContractIds;
+                    } else {
```

Comment:
Always felt split between the two different styles, because while it's true you reduce the levels of indentation, it's also true that you end up with two different returns, while with the other style you have only one place (the end of the function) where any value is returned and thus maybe the code flow is easier to follow